crates/sui/src/validator_commands.rs

```diff
@@ -0,0 +1,342 @@
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.worker_key().as_bytes().to_vec()).unwrap(),
+                    ),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator_info.proof_of_possession.as_ref().to_vec())
+                            .unwrap(),
+                    ),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.name().to_owned().into_bytes()).unwrap(),
+                    ),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.description.clone().into_bytes()).unwrap(),
+                    ),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.image_url.clone().into_bytes()).unwrap(),
+                    ),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.project_url.clone().into_bytes()).unwrap(),
+                    ),
+                    CallArg::Pure(bcs::to_bytes(&validator.network_address().to_vec()).unwrap()),
+                    CallArg::Pure(bcs::to_bytes(&validator.p2p_address().to_vec()).unwrap()),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.narwhal_primary_address().to_vec()).unwrap(),
+                    ),
+                    CallArg::Pure(
+                        bcs::to_bytes(&validator.narwhal_worker_address().to_vec()).unwrap(),
+                    ),
+                    CallArg::Pure(bcs::to_bytes(&validator.gas_price()).unwrap()),
+                    CallArg::Pure(bcs::to_bytes(&validator.commission_rate()).unwrap()),
+                ];
+                let response =
+                    call_0x5(context, "request_add_validator_candidate", args, &client).await?;
+                SuiValidatorCommandResponse::BecomeCandidate(response)
+            }
+
+            SuiValidatorCommand::JoinCommittee => {
+                let response = call_0x5(context, "request_add_validator", vec![], &client).await?;
+                SuiValidatorCommandResponse::JoinCommittee(response)
+            }
+
+            SuiValidatorCommand::LeaveCommittee => {
+                let response =
+                    call_0x5(context, "request_remove_validator", vec![], &client).await?;
+                SuiValidatorCommandResponse::LeaveCommittee(response)
+            }
+        });
+        ret
+    }
+}
+
+async fn call_0x5(
+    context: &mut WalletContext,
+    function: &'static str,
+    call_args: Vec<CallArg>,
+    sui_client: &SuiClient,
+) -> anyhow::Result<SuiTransactionResponse> {
+    let sender = context.active_address()?;
+    let gas_obj_ref = get_gas_obj_ref(sender, sui_client).await?;
+    let mut args = vec![SUI_SYSTEM_OBJ_CALL_ARG];
+    args.extend(call_args);
+    let rgp = sui_client
+        .governance_api()
+        .get_reference_gas_price()
+        .await?;
+    // 10k is a herustic number for gas unit
+    let gas_budget = 10_000 * rgp;
```

Comment:
Do we want to have a cli arg to override the gas_budget? Not necessarily needed now but may be useful addition later.

Reply:
I tried transaction builder but sadly found `SuiJsonValue` very challenging to accommodate vectors of bytes. Perhaps I just don't know how to use it

Reply:
yeah we should have that. also need to update the gas selection function